### PR TITLE
[Optimization](functions)Optimize function call for const columns.

### PR DESCRIPTION
**Pull request by zclllyybb**
## Problem summary

Continue to do optimization that https://github.com/apache/doris/pull/18212 did some of before.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/functions/in.h [hunk 1]
```diff
@@ -227,6 +187,61 @@ class FunctionIn : public IFunction {
...
+            in_state->hybrid_set->find_batch_negative(*col_ptr, input_rows_count, vec_res);
+        }
+    }
+    static void impl_without_set(FunctionContext* context,
+                                 const std::vector<ColumnPtr>& set_columns, size_t input_rows_count,
+                                 ColumnUInt8::Container& vec_res,
+                                 ColumnUInt8::Container& vec_null_map_to,
+                                 const ColumnPtr& materialized_column, bool col_const) {
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            const auto& ref_data = materialized_column->get_data_at(col_const ? 0 : i);
```
Comment (HappenLee) [inline be/src/vec/functions/in.h:219]: not reference

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/functions/array/function_array_set.h [hunk 2]
```diff
@@ -216,6 +213,88 @@ struct ArraySetImpl {
...
+        constexpr auto execute_left_column_first = Impl::Action::execute_left_column_first;
+        size_t current = 0;
+        Impl impl;
+        size_t left_off = (*left_data.offsets_ptr)[-1];
+        size_t left_len = (*left_data.offsets_ptr)[0] - left_off;
+        for (size_t row = 0; row < left_data.offsets_ptr->size(); ++row) {
+            size_t count = 0;
+            size_t right_off = (*right_data.offsets_ptr)[row - 1];
+            size_t right_len = (*right_data.offsets_ptr)[row] - right_off;
+            if constexpr (execute_left_column_first) {
```
Comment (HappenLee) [inline be/src/vec/functions/array/function_array_set.h:233]:
```
if constexpr (execute_left_column_first) {
                impl.template apply<true>(left_data, left_off, left_len, dst, &count);
                impl.template apply<false>(right_data, right_off, right_len, dst, &count);
            } else {
                impl.template apply<false>(right_data, right_off, right_len, dst, &count);
                impl.template apply<true>(left_data, left_off, left_len, dst, &count);
            }
            current += count;
            dst.offsets_ptr->push_back(current);
            impl.reset();
```
the code better be a template func

File: be/src/vec/functions/function_conv.cpp [hunk 3]
```diff
@@ -49,58 +52,99 @@ class FunctionConv : public IFunction {
...
+                                    const Int8 src_base, const Int8 dst_base,
+                                    ColumnString* result_column, NullMap& result_null_map,
+                                    size_t input_rows_count) {
+        if (std::abs(src_base) < MathFunctions::MIN_BASE ||
+            std::abs(src_base) > MathFunctions::MAX_BASE ||
+            std::abs(dst_base) < MathFunctions::MIN_BASE ||
+            std::abs(dst_base) > MathFunctions::MAX_BASE) { //out of bound
+            for (int i = 0; i < input_rows_count; i++) {
                 result_null_map[i] = true;
                 result_column->insert_default();
```
Comment (HappenLee) [inline be/src/vec/functions/function_conv.cpp:139]: do not use for loop
```
memset->result_null_map
result_column->insert_many_default
```

Comment (HappenLee) [inline be/src/vec/functions/function_conv.cpp:139]: do not use for loop
```
memset->result_null_map
result_column->insert_many_default
```

File: be/src/vec/functions/function_json.cpp [hunk 4]
```diff
@@ -244,17 +244,44 @@ struct GetJsonNumberType {
         for (size_t i = 0; i < size; ++i) {
             const char* l_raw_str = reinterpret_cast<const char*>(&ldata[loffsets[i - 1]]);
             int l_str_size = loffsets[i] - loffsets[i - 1];
-
```
Comment (HappenLee) [inline be/src/vec/functions/function_json.cpp:247]: better use bool do do check

Comment (zclllyybb) [reply be/src/vec/functions/function_json.cpp:247]: it has other common dependencies. found another way to extract common logic.

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run p0

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**